### ISSUE TITLE
[Classic] Crash when taming a rare pet

### ISSUE THREAD

**Issue by time-out** [cmangos/issues#551]
The server crashes when a hunter is trying to tame a rare pet like Grunter/Suhlaman (creature id 8303).
It also happened with npc 2753, 2850 and 14222. [GDB report here...](https://gist.github.com/time-out/1ddcf6c6697202510a66)
I hope the backtrace is sufficient. I'm not so familiar with the gdb stuff. ;p

--
CMaNGOS Classic/ (* * Revision 2583 - f41805695ff..) for Linux_x64 (little-endian)
Using script library: ScriptDev2 (for CMaNGOS classic z2451+) Revision z2715 (f41805695ff97ac9992cb25df72011a72ba7bf0c) * * (Unix)
Using World DB: Classic DB version 1.4.2 'High Elves, Thaurissan and Necromancers in a Swimming Pool'. For core z2583 and SD2 z2715
Using creature EventAI: ACID 1.12.x (Classic) for CMaNGOS-CLASSIC (1.12.X Client)

**Comment by Schmoozerd** [comment]
i tested it with a freshly spawned 8303 on wotlk and got no problem.
Can you check if the npc you tried to tame has some auras on spawn in creature_addon table?

**Comment by time-out** [comment]
```
mysql> SELECT * FROM creature AS a LEFT OUTER JOIN creature_addon AS b ON b.guid=a.guid WHERE a.id=8303;
+--------+------+-----+---------+--------------+------------+------------+------------+-------------+---------------+-----------+-----------------+-----------+---------+------------+--------------+------+-------+--------+-------------+------------+-------+-----------+-------+
| guid   | id   | map | modelid | equipment_id | position_x | position_y | position_z | orientation | spawntimesecs | spawndist | currentwaypoint | curhealth | curmana | DeathState | MovementType | guid | mount | bytes1 | b2_0_sheath | b2_1_flags | emote | moveflags | auras |
+--------+------+-----+---------+--------------+------------+------------+------------+-------------+---------------+-----------+-----------------+-----------+---------+------------+--------------+------+-------+--------+-------------+------------+-------+-----------+-------+
|  45166 | 8303 |   0 |    8870 |            0 |   -11261.9 |   -3120.39 |       2.87 |     3.83534 |         72000 |         0 |               0 |      2769 |       0 |          0 |            0 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134324 | 8303 |   0 |       0 |            0 |   -11211.7 |   -2995.63 |    3.36572 |     4.60202 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134325 | 8303 |   0 |       0 |            0 |   -11178.3 |   -2992.61 |    8.94092 |     4.26665 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134326 | 8303 |   0 |       0 |            0 |   -11204.8 |      -3055 |    2.42608 |     4.72061 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134327 | 8303 |   0 |       0 |            0 |   -11247.5 |    -3079.9 |    3.29563 |     5.47852 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134328 | 8303 |   0 |       0 |            0 |   -11200.8 |   -3103.99 |    8.13294 |     5.33951 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134329 | 8303 |   0 |       0 |            0 |     -11240 |    -3172.6 |    2.69219 |     4.49206 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134330 | 8303 |   0 |       0 |            0 |   -11183.3 |   -3174.74 |    6.94442 |  0.00194143 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134331 | 8303 |   0 |       0 |            0 |   -11150.5 |   -3222.62 |    7.76973 |     5.39763 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134332 | 8303 |   0 |       0 |            0 |     -11148 |   -3268.85 |    9.20335 |     4.77559 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134333 | 8303 |   0 |       0 |            0 |   -11196.2 |   -3297.75 |    9.86134 |        3.57 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134334 | 8303 |   0 |       0 |            0 |   -11163.2 |   -3328.39 |    5.48639 |     3.90458 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134335 | 8303 |   0 |       0 |            0 |   -11194.4 |   -3361.09 |    8.73583 |     3.56372 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134336 | 8303 |   0 |       0 |            0 |   -11232.5 |   -3336.23 |    9.21853 |     3.25899 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134337 | 8303 |   0 |       0 |            0 |   -11204.5 |   -3227.08 |    9.37905 |     1.82563 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134338 | 8303 |   0 |       0 |            0 |   -11205.1 |   -3145.22 |    7.78832 |     2.08089 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134339 | 8303 |   0 |       0 |            0 |   -11308.1 |   -3153.34 |    6.07638 |     3.62577 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
| 134340 | 8303 |   0 |       0 |            0 |   -11332.3 |   -3119.14 |   -1.65414 |     1.88375 |        151200 |        15 |               0 |      2769 |       0 |          0 |            1 | NULL |  NULL |   NULL |        NULL |       NULL |  NULL |      NULL | NULL  |
+--------+------+-----+---------+--------------+------------+------------+------------+-------------+---------------+-----------+-----------------+-----------+---------+------------+--------------+------+-------+--------+-------------+------------+-------+-----------+-------+
```

**Comment by Patman64** [comment]
Hi. Thanks for the crash report.

If you are still experiencing issues please make sure you build in **debug mode** (and don't `strip` the binaries) and use `backtrace full`. There's not much information in the log you posted.

**Comment by time-out** [comment]
See: [BACKTRACE FULL](https://gist.github.com/time-out/1ddcf6c6697202510a66)

**Comment by cyberium** [comment]
Sorry cannot reproduce too -> http://i.imgur.com/Ro2vhHV.jpg. Iam on window btw. So maybe another one with linux version could confirm/infirm this?

**Comment by N8boy** [comment]
I can confrim this. On the classic server hosted on a linux systems from time to time the rare-pets causes a crash of the client side, you will be kicked/logged out. 
Sadly I can't post anylogs because I don't have the access to the logs

**Comment by Phatcat** [comment]
Solved by @Caydan 's commit  in Pet patch 1.1 based on @H0zen 's commit to MangosZero.
